Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py

```diff
@@ -0,0 +1,601 @@
+        if key == 'resolution_status':
+            filters['resolutionStatus'] = {'eq': RESOLUTION_STATUS_OPTIONS[value]}
+        if key == 'file_type':
+            filters['fileType'] = {'eq': FILE_TYPE_OPTIONS[value]}
+        if key == 'sharing':
+            filters[key] = {'eq': FILE_SHARING_OPTIONS[value]}
+        if key == 'extension':
+            filters[key] = {'eq': value}
+        if key == 'quarantined':
+            filters[key] = {'eq': argToBoolean(value)}
+        if key == 'type':
+            filters[key] = {'eq': value}
+        if key == 'group_id':
+            filters['userGroups'] = {'eq': value}
+        if key == 'is_admin':
+            filters['isAdmin'] = {'eq': value}
+        if key == 'is_external':
+            filters['isExternal'] = {'eq': IS_EXTERNAL_OPTIONS[value]}
+        if key == 'status':
+            filters[key] = {'eq': STATUS_OPTIONS[value]}
+
+        request_data['filters'] = filters
+    return request_data
+
+
+def build_filter_and_url_to_search_with(url_suffix, customer_filters, arguments, command_id=''):
+    request_data = {}
+    if command_id:
+        url_suffix += command_id
+    elif customer_filters:
+        request_data = json.loads(customer_filters)
+    else:
+        request_data = args_to_filter(arguments, url_suffix)
+    return request_data, url_suffix
+
+
+def args_to_filter_dismiss_alerts_and_resolve_alerts(alert_ids, customer_filters, comments):
+    request_data = {}
+    filters = {}
+    if alert_ids:
+        ids = {'eq': alert_ids.split(',')}
+        filters['id'] = ids
+        if comments:
+            request_data['comment'] = comments
+        request_data['filters'] = filters
+    elif customer_filters:
+        request_data = json.loads(customer_filters)
+    return request_data
+
+
+def params_to_filter(parameters):
+    """
+    Turns the parameters to filters.
+    Args:
+        parameters: The parameters that should to be filters.
+    Returns:
+        The filter we built using the parameters.
+    """
+    filters = {}
+    if 'severity' in parameters.keys():
+        filters['severity'] = {'eq': SEVERITY_OPTIONS[parameters['severity']]}
+    if 'resolution_status' in parameters.keys():
+        filters['resolutionStatus'] = {'eq': RESOLUTION_STATUS_OPTIONS[parameters['resolution_status']]}
+    if 'service' in parameters.keys():
+        filters['entity.service'] = {'eq': (int(parameters['service']))}
+    if 'instance' in parameters.keys():
+        filters['entity.instance'] = {'eq': (int(parameters['instance']))}
+    return filters
+
+
+def test_module(client):
+    try:
+        client.list_alerts(url_suffix='/alerts/', request_data={})
+        if demisto.params().get('isFetch'):
+            client.list_incidents(filters={}, limit=1)
+    except DemistoException as e:
+        return str(e)
+    return 'ok'
+
+
+def alerts_output_to_readable_output(alerts):
+    if 'data' not in alerts:
+        alerts = {'data': [alerts]}
```

Comment:
In most cases the data that comes back from the api is in a dictionary called "data", in a small number of cases this is not the case, so in case it is not in a dictionary "data"  I put it in a dictionary.